src/control/settings/Settings.cpp

```diff
@@ -596,6 +598,15 @@ auto Settings::load() -> bool {
     loadButtonConfig();
     loadDeviceClasses();
 
+    paletteFilePath = PALETTE_FILE;
+    auto paletteFile = Util::getConfigFile(paletteFilePath);
+    if (!fs::exists(paletteFile)) {
+        Palette::create_default(paletteFile);
+    }
+    this->palette = new Palette(std::move(paletteFile));
+    this->palette->load();
```

Comment:
**Question:** is there a failover, if the file couldn't be load?

Reply:
Yes, the the topmost line in your snippet calls it.

Basically the behaviour is:
1. Try to read palette file in `.config/xournalpp` foler (or similar)
      a. if file does not exist create default palette file at specified path from xournalpp colorscheme (useful for soft migration)
2. Parse file

The only case that is not caught is when the palette file exists but has a non parseable format. For this case we still should define the expected behaviour.

I'd propose in case of exception or empty palette (no colors). Show warning and use default color scheme. What do you think?

Reply:
> The only case that is not caught is when the palette file exists but has a non parseable format. For this case we still should define the expected behaviour.

Yes I would use the same behavior, override the file and use the default palette.

But it may be better to just load the default settings from constexpr variables. And then save the file.

```
// try {
color_palatte.load();
if (<onfail>) /* } catch(std::exeption const& ...) */ {
    color_palatte = generate_default_palette();
    color_palatte.save()
}

```

Reply:
I though about overriding at first as well. But I think that would annoy users quite a bit.

Imagine you tweaked your palette but did a simple formatting mistake. You start xournalpp. Upps, now your palette file is overwritten with the default :see_no_evil: 

I'd rather show a "warning" message with `g_message` and load the default palette but not overwrite the palette file.

Reply:
A dialog would be even better. g_messages are quite hard to get to on Windows.

Reply:
Good idea I'll try that out.